src/main/java/com/easypost/service/BetaReferralCustomerService.java

```diff
@@ -0,0 +1,89 @@
+package com.easypost.service;
+
+import java.util.HashMap;
+
+import com.easypost.exception.EasyPostException;
+import com.easypost.http.Requestor;
+import com.easypost.http.Requestor.RequestMethod;
+import com.easypost.model.BetaPaymentRefund;
+import com.easypost.model.PaymentMethod;
+import com.easypost.model.PaymentMethodObject;
+
+public class BetaReferralCustomerService {
+    private final EasyPostClient client;
+
+    /**
+     * BetaReferralCustomerService constructor.
+     * 
+     * @param client The client object.
+     */
+    BetaReferralCustomerService(EasyPostClient client) {
+        this.client = client;
+    }
+
+    /**
+     * Add Stripe payment method to referral customer.
+     *
+     * @param stripeCustomerId       ID of the Stripe account.
+     * @param paymentMethodReference Reference of Stripe payment method.
+     * @return PaymentMethodObject object.
+     * @throws EasyPostException
+     */
+    public PaymentMethodObject addPaymentMethod(String stripeCustomerId, String paymentMethodReference)
+            throws EasyPostException {
+        return addPaymentMethod(stripeCustomerId, paymentMethodReference, PaymentMethod.Priority.PRIMARY);
+    }
+
+    /**
+     * Add Stripe payment method to referral customer.
+     *
+     * @param stripeCustomerId       ID of the Stripe account.
+     * @param paymentMethodReference Reference of Stripe payment method.
+     * @param priority               Priority of this payment method.
+     * @return PaymentMethodObject object.
+     * @throws EasyPostException
+     */
+    public PaymentMethodObject addPaymentMethod(String stripeCustomerId, String paymentMethodReference,
+            PaymentMethod.Priority priority) throws EasyPostException {
+        HashMap<String, Object> params = new HashMap<>();
+        params.put("stripe_customer_id", stripeCustomerId);
+        params.put("payment_method_reference", paymentMethodReference);
+        params.put("priority", priority);
+
+        HashMap<String, Object> wrappedParams = new HashMap<>();
+        wrappedParams.put("payment_method", params);
+
+        return Requestor.request(RequestMethod.POST, "%s/beta/referral_customers/payment_method", wrappedParams,
```

Comment:
This doesn't use the Referral API key to make the request.

Reply:
Per our discussion, this is set up correctly. You would have to use a EasyPostClient with `referral`